### PR TITLE
apply patch from pr #707

### PR DESCRIPTION
**Pull request by dhawal1248**

### CODE REVIEW

File: database/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -193,7 +193,13 @@ func (ch *ClickHouse) SetVersion(version int, dirty bool) error {
...
 	}
 
-	query := "INSERT INTO " + ch.config.MigrationsTable + " (version, dirty, sequence) VALUES (?, ?, ?)"
+	query := fmt.Sprintf(
+		"INSERT INTO %s (version, dirty, sequence) VALUES (%d, %d, %d)",
+		ch.config.MigrationsTable,
+		version,
+		bool(dirty),
+		time.Now().UnixNano(),
+	)
```
Comment (srikanthccv) [inline database/clickhouse/clickhouse.go:202]: I am trying to understand what does this fix? Why was there a need to change this? There is a line below 

```
tx.Exec(query, version, bool(dirty), time.Now().UnixNano())
```
as we can see, the values are passed as args to query. So, I imagine the original query with args should work?

Comment (dhawal1248) [reply database/clickhouse/clickhouse.go:202]: Yeah i was also thinking the same..that is why i asked @nityanandagohain to also take a look if we need this

Comment (nityanandagohain) [reply database/clickhouse/clickhouse.go:202]: I didn't actually get what we are trying to do here

Comment (dhawal1248) [reply database/clickhouse/clickhouse.go:202]: @nityanandagohain we are currently using a fork of golang-migrate in signoz-otel-collector, we use [this](https://github.com/sergey-telpuk/migrate/tree/master) forked repo
This pr only patches the changes in that forked repo to this repo.

Comment (dhawal1248) [reply database/clickhouse/clickhouse.go:202]: https://github.com/SigNoz/signoz-otel-collector/blob/3a7486b2329a317aaab8c3c4c55b0f11fdda855a/go.mod#L490

Comment (srikanthccv) [reply database/clickhouse/clickhouse.go:202]: I looked at the full git history. It wasn't Nitya who added it. @makeavish I see this commit is originally added this https://github.com/SigNoz/opentelemetry-collector-contrib/commit/bd45c958992b7d23e4369cf39d9172aa6e64890d and the this is the diff Dhawal applied here https://github.com/golang-migrate/migrate/compare/master...sergey-telpuk:migrate:master. Can you help us understand what was it trying to fix?

Comment (dhawal1248) [reply database/clickhouse/clickhouse.go:202]: @srikanthccv @makeavish maybe https://github.com/golang-migrate/migrate/pull/723 is a better way to solve this? this uses formatted strings whereas the pr i shared uses prepared statements which are safer

Comment (makeavish) [reply database/clickhouse/clickhouse.go:202]: @dhawal1248 @srikanthccv go-migrate uses clickhouse-go v1.4 and was not updated to clickhouse-go v2. The go-migrate maintainers have been very slow to review so I had to use fork while clickhouse-go v2 support was being added but I see it's still not added.

Comment (makeavish) [reply database/clickhouse/clickhouse.go:202]: Here's another [PR](https://github.com/golang-migrate/migrate/pull/697) from clickhouse-go maintainer which is not merged.

**Pull request merged**